### PR TITLE
Add list_repo_likers method to HfApi

### LINKED ISSUE #1705: Return the liked users for for a huggingface model
Currently, the is a function "list_liked_repos" to list user-liked models by inputting the user name. It would be better to have a function that outputs the list of liked users for a given model.

### PR DESCRIPTION
**Pull request by issamarabi**
### Summary
This PR introduces the `list_repo_likers` method to the `HfApi` class, allowing users to retrieve a list of likers for a given model repository on the Hugging Face Hub.

### Changes
- Implemented `list_repo_likers` method in `HfApi`.
- Added corresponding dataclass `User` to represent users who liked a repo.
- Integrated a new API endpoint to fetch likers for a given model repository.
- Added unit tests to ensure the correctness of the method.

### Motivation
- Fixes #1705 - Return the liked users for a Hugging Face model
- Users will now have the ability to see who liked their models, fostering community interactions and recognition.

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -1730,6 +1757,51 @@ def list_liked_repos(
...
+    @validate_hf_hub_args
+    def list_repo_likers(
+        self,
+        repo_id: str,
+        *,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> List[User]:
+        """
+        List all users who liked a given repo on huggingface.co.
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:1769]:
```suggestion
        List all users who liked a given repo on the hugging Face Hub.
```

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -1730,6 +1757,51 @@ def list_liked_repos(
...
+
+        # Make the request
+        response = get_session().get(path, headers=headers)
+        hf_raise_for_status(response)
+
+        # Parse the results into User objects
+        likers_data = response.json()
+        likers = [User(**user_data) for user_data in likers_data]
+
+        return likers
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:1803]:
```suggestion
        return [User(**user_data) for user_data in likers_data]
```

**Review by Wauplin — APPROVED** [review]
Very clean and complete PR, thanks @issamarabi! I added 2 minor comments but overall looks good. :tada:

**Review by Wauplin — APPROVED** [review]
Hey @issamarabi thanks for making the changes! I re-reviewed and tested it locally. I hope you don't mind, I pushed a commit to your branch to:
- apply formatting (run `make style`)
- add `User` object to the docs (I forgot to mention it)
- add `User` and `list_repo_likers` at root level. This makes them usable without intanciating `HfApi` yourself
- fix an attribute name

I think we are good to merge now. Great addition!

**Comment by julien-c** [comment]
awesome job @issamarabi 🔥

**Pull request merged**